src/main/java/org/wise/portal/service/usertags/impl/UserTagsServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package org.wise.portal.service.usertags.impl;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.hibernate.proxy.HibernateProxyHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.acls.domain.ObjectIdentityImpl;
+import org.springframework.security.acls.model.ObjectIdentity;
+import org.springframework.stereotype.Service;
+import org.wise.portal.dao.authentication.AclTargetObjectIdentityDao;
+import org.wise.portal.dao.usertags.UserTagsDao;
+import org.wise.portal.domain.authentication.MutableAclTargetObjectIdentity;
+import org.wise.portal.domain.project.Project;
+import org.wise.portal.domain.user.User;
+import org.wise.portal.domain.usertag.UserTag;
+import org.wise.portal.domain.usertag.impl.UserTagImpl;
+import org.wise.portal.service.usertags.UserTagsService;
+
+@Service
+public class UserTagsServiceImpl implements UserTagsService {
+
+  @Autowired
+  private AclTargetObjectIdentityDao<MutableAclTargetObjectIdentity> aclTargetObjectIdentityDao;
+
+  @Autowired
+  private UserTagsDao<UserTag> userTagsDao;
+
+  @Override
+  public UserTag get(User user, String text) {
+    return userTagsDao.get(user, text);
+  }
+
+  @Override
+  public UserTag createTag(User user, String text) {
+    UserTag userTag = new UserTagImpl(user, text);
+    userTagsDao.save(userTag);
+    return userTag;
+  }
+
+  @Override
+  public Boolean isOwner(User user, Long tagId) {
+    UserTag userTag = userTagsDao.get(tagId);
+    return userTag != null && userTag.getUser().getId().equals(user.getId());
+  }
+
+  @Override
+  public UserTag editTag(Long tagId, String tag) {
+    UserTag userTag = userTagsDao.get(tagId);
+    userTag.setText(tag);
+    userTagsDao.save(userTag);
+    return null;
+  }
```

Comment:
Looks like this method is never called. Remove from class and UserTagsService interface?